eloquentjs/5_4_dominant_writing_direction.js

```diff
@@ -0,0 +1,118 @@
+const SCRIPTS = require('./bin/scripts.js');
+
+/**
+ * Returns a string indicating the dominant writing direction present in an input string. 
+ * If there is a tie between dominant writing directions its resolved by alphabetic ordering of direction strings(ltr < rtl < ttb).
+ * @param {*} inputString 
+ * @returns 
+ */
+function dominantWritingDirection(input) {
+    input = (input.split('').filter(b => scriptFromCharacter(b) !== null)).join(''); // filters out invalid characters
+
+    let directionGroups = countBy(input, j => scriptFromCharacter(j).direction);
+
+    directionGroups.sort((a, b) => {
+        if (a.count < b.count) {
+            return 1;
+        }
```

Comment:
Using else statements would make the flow of this a bit more visually obvious and also make it take up less space